src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/Authentication/CookieExtensions.cs

```diff
@@ -0,0 +1,28 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Owin.Security.Interop;
+
+namespace Microsoft.Owin.Security.Cookies;
+
+public static class CookieExtensions
+{
+    public static void ConfigureSharedCookie(this CookieAuthenticationOptions options,
+        string sharedApplicationName,
+        string authScheme,
+        string sharedKeyDirectory)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        var sharedDataProtectionProvider = DataProtectionProvider.Create(
```

Comment:
Rather than call `DataProtectionProvider.Create` here, it might be better to have the user provide a data protection provider since they may not want to use a shared directory for this (maybe Redis or some other solution works better in their scenario).

Though if we're removing that part of what the extension method does, maybe it does make sense to just address this with docs and samples as suggested in other comments.